src/components/TransactionDetails/TransactionDetails.jsx

```diff
@@ -6,50 +6,56 @@ import { transactionsTotalsSelector } from '../../redux/modules/transactionsList
 
 import NavLink from '../NavLinks/NavLink.jsx';
 import TransactionDetailsList from './TransactionDetailsList.jsx';
+import Spinner from '../Spinner/Spinner.jsx';
 
 import { HOME, TRANSACTIONS } from '../../routes';
 
 const TransactionDetails = ({ location }) => {
   const list = useSelector(transactionsTotalsSelector);
 
+  if (list.length < 1) {
+    return <Spinner />;
+  }
+
```

Comment:
oh, you use this checking three times (first one - here, second - BalancePage and third - in TransactionsListItem), I am thinking about the wrapper component with the following content:
```
const TransactionsWrapper = ({ children, list }) => {
const isListEmpty = list.length === 0

if (listIsEmpty) {
  return <Spinner />;
}

return children;
}
```

then use it here lnside return function
```
<TransactionsWrapper>
  <div className="transactions_list__page">
    ....
  </div>
</TransactionsWrapper>
```